ls1intum/Artemis
# Adjust date assignment logic

### Checklist

### Motivation and Context


This behaviour occured once here (#1535 ), which was the main motiviation for this overall change. 
This adjustment solves the problem of undefined dates being accidentally evaluated to moment(undefined) which equals moment() and therefore now. A comparison to null is not needed, since if it's null it's evluated to false anyways. Thanks @filip-gregurevic for pointing that out.
### Description
I changed the general assignments of dates from 
`res.body.submittedTime = res.body.startDate!= null ? moment(res.body.startDate) : null;` 
to 
`res.body.submittedTime = res.body.startDate? moment(res.body.startDate) : null; `
e.g..
Aswell, I removed some unused methods and reused a few methods which were implemented identically in two services.

### Steps for Testing


1. Log in to Artemis
2. Check whether all dates are still shown correctly
